internal/webhooks/inclusterippool.go

```diff
@@ -191,18 +191,36 @@ func validateAddresses(newPool types.GenericInClusterPool) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "prefix"), newPool.PoolSpec().Prefix, "a valid prefix is required when using addresses"))
 	}
 
-	if newPool.PoolSpec().Gateway != "" {
-		_, err := netip.ParseAddr(newPool.PoolSpec().Gateway)
-		if err != nil {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "gateway"), newPool.PoolSpec().Gateway, err.Error()))
+	for _, address := range newPool.PoolSpec().Addresses {
+		if !poolutil.AddressStrParses(address) {
```

Comment:
Tbf this is just a performance nit, but can we get rid of `AddressStrParses` and just check for the error in the second loop below? Currently we're parsing each entry twice, that's not really necessary.

Reply:
That makes sense, updated.